cli/js/testing.ts

```diff
@@ -110,7 +228,56 @@ function filterTests(
     }
 
     return passes;
-  });
+  };
+}
+
+interface TestReporter {
+  start(msg: StartMsg): Promise<void>;
+  test(msg: TestMsg): Promise<void>;
+  end(msg: EndMsg): Promise<void>;
+}
+
+class ConsoleReporter implements TestReporter {
+  constructor(private console: Console) {}
+
+  async start(msg: StartMsg): Promise<void> {
```

Comment:
@ry for `cli/js/tests` I'd swap this out for `FileReporter` and pass it as argument to `runTests({ ... })`